components/core/src/os/process/linux.rs

```diff
@@ -54,3 +52,116 @@ fn become_exec_command(command: PathBuf, args: Vec<OsString>) -> Result<()> {
     // failed to exec to our target program
     return Err(error_if_failed.into());
 }
+
+pub struct Child {
+    pid: u32
+}
+
+impl Child {
+    pub fn new(child: &mut process::Child) -> Child {
+        Child { pid: child.id() }
+    }
+
+    pub fn id(&self) -> u32 {
+        self.pid
+    }
+
+    pub fn status(&mut self) -> Result<HabExitStatus> {
+        let mut exit_status: i32 = 0;
+
+        match unsafe { waitpid(self.pid as i32, &mut exit_status, 1 as libc::c_int) } {
+            0 => Ok(HabExitStatus { status: None }),
+            -1 => Err(Error::WaitpidFailed(format!("Error calling waitpid on pid: {}", self.pid))),
+            _ => Ok(HabExitStatus { status: Some(exit_status as u32) })
+        }
+    }
+}
+
+impl ExitStatusExt for HabExitStatus {
+    fn code(&self) -> Option<u32> {
+        unsafe {
```

Comment:
It doesn't appear that you aren't performing any unsafe operations in this code block. You should remove the unsafe block here if that's the case!